lib/modules/datasource/packagist/schema.ts

```diff
@@ -144,3 +144,82 @@ export function parsePackagesResponses(
 
   return result;
 }
+
+const RegistryFile = z.object({
+  key: z.string(),
+  sha256: z.string(),
+});
+export type RegistryFile = z.infer<typeof RegistryFile>;
+
+const RegistryMetaFile = z.object({
+  sha256: z.string().nullable(),
+});
+type RegistryMetaFile = z.infer<typeof RegistryMetaFile>;
+
+const RegistryMetaFiles = z
+  .record(RegistryMetaFile.nullable().catch(null))
+  .transform((obj) => {
+    // Remove all null values
+    // TODO: extract as schema utility
+    const result: Record<string, RegistryMetaFile> = {};
+    for (const [key, val] of Object.entries(obj)) {
+      if (val !== null) {
+        result[key] = val;
+      }
+    }
+    return result;
+  });
+
+export const RegistryMeta = z
+  .object({
+    ['includes']: RegistryMetaFiles.transform((obj): RegistryFile[] => {
```

Comment:
This object key syntax used intentionally for the uniformity. It will be eliminated in further refactoring.